### ISSUE TITLE
Modifying the band distribution in Vocoder

### ISSUE THREAD

**Issue by unfa** [calf-studio-gear/calf#105]
Hearning what Image-Line Vocodex can do inspired me to seek for a way to really mess up the vocoding process.

Right now Calf Vocoder always uses a fixed, even distribution of the filters - there's no way to alter their position.

I'd like to propose a "Band distribution" section to be added that'd allow the user to change the filter bank distribution parametrically with a few functions.

Defining the vocoded frequency range:
- Min - frequency of the lowest band
- Max - frequency of the highest band

- Tilt - makes the bands more close to each other in the top or the bottom part of the spectrum. Positive values make  the top frequencies have more bands, while low have less, negative does the opposite.

The image below illustrates how tilt would affect the band distribution. Top is 0 tilt, middle is positive tilt, and the bottom is negative tilt:
![selection_055](https://cloud.githubusercontent.com/assets/491704/21046303/5a6af040-be04-11e6-8f76-c2b264b26ee0.png)

- Peak center - defines the center frequency of the peak
- Peak width - defines the size of the affected range (the width of the gaussian curve that defines band offset)
- Peak amount - 0 means no change, 1 means the frequencies are spread wider (expanded outwards, away from the peak center)

The image below illustrates what would the peak do to the band distribution. The top image is unaffected, the middle is a positive peak (expanding) and the bottom is negative peak (condensing)

![selection_054](https://cloud.githubusercontent.com/assets/491704/21046169/9bfe2406-be03-11e6-9558-b92c546b362f.png)

Apart from this a "Balance Q" parameter could be used. When at 100% it'd make bands wider in places where there is less bands, and make the Q narrower where there is more bands - to keep the coverage balanced, and avoid creating holes or overlaps (unless the user wants that) - it should probably by on by default.

So we have 3 functions to modify the band distribution.
By allowing very narrow Q values (leaving gaps between the bands) and automating the min, max, tilt and peak values - Calf Vocoder should be able to create some very interesting sounds, not heard before in the linux audio world.

What do you think?

![selection_056](https://cloud.githubusercontent.com/assets/491704/21046652/f52168ac-be05-11e6-82e8-c51511989106.png)

**Comment by unfa** [comment]
This is like the fastest feature implementation I've ever seen :D Thank you, @boomshop !
I can't wait to test this out!

**Comment by unfa** [comment]
I tested the new features yesterday, and along with the new High Q (which sounds awesome!) a whole pallet of new exciting sounds can be made. Neurofunk producers should be satisfied.

There's a small problem with changing the Low, High and Tilt values in realtime - the bands seem to shift their output frequency, but their input frequency seems to lag behind and update only after I release my mouse drag. This needs some dedicated investigation, so I'll probably report a separate issue.

I might make a video about this update in the future to show people that it's here.

**Comment by boomshop** [comment]
That sounds odd cause _all_ filters are set at the very same moment - as soon as any filter related value changes. There's no chance that a detector filter has a different setting than the corresponding modulator filter. Could you please explain/show off how you came to that conclusion?
